Error fixes

besarism opened the pull request.

besarism commented on TextFieldEffects/TextFieldEffects/MadokaTextField.swift:
```diff
@@ -183,12 +183,13 @@ import UIKit
+//// Helper function inserted by Swift 4.2 migrator.
+//fileprivate func convertFromCAShapeLayerLineCap(_ input: CAShapeLayer) -> String {
+//    return
+//}
```
@raulriera done

besarism commented:
@raulriera yeah you were right I was using swift4 and not swift4.2 anyways I fixed errors that I was facing. Check it out now.

besarism commented:
@raulriera yeah you were right, I was using swift4 instead of swift4.2, I adapted the code to latest swift and I fixed the errors that I was facing, check it out now.

raulriera commented:
Are you sure these changes are needed? I get no errors compiling the project, can you test the latest version in master? What errors are you getting and can you reproduce them on a sample project?

besarism commented:
> Are you sure these changes are needed? I get no errors compiling the project, can you test the latest version in master? What errors are you getting and can you reproduce them on a sample project?

Hi @raulriera 
 
 borderLayer.lineCap = convertToCAShapeLayerLineCap(convertFromCAShapeLayerLineCap(CAShapeLayerLineCap.square)) (line 118) - error: Type 'CAShapeLayerLineCap' (aka 'NSString') has no member 'square'


return CAShapeLayerLineCap(rawValue: input) (line 188) - error: Argument labels '(rawValue:)' do not match any available overloads

return input.rawValue (line 193) - error: Value of type 'CAShapeLayerLineCap' (aka 'NSString') has no member 'rawValue'


<img width="1440" alt="screen shot 2018-10-09 at 8 32 33 am" src="https://user-images.githubusercontent.com/34536227/46650790-2755b600-cb9e-11e8-86db-4453d41d8a05.png">



<img width="1083" alt="screen shot 2018-10-09 at 8 35 04 am" src="https://user-images.githubusercontent.com/34536227/46650830-45231b00-cb9e-11e8-98bc-cc158938108e.png">


So the only fixes after I run it in swift 4.2 are on this file (MadokaTextField), the other files that I commited earlier are just short way of your code (and there is no problem which one you choose).
btw. I am using TextFieldEffects on one of my clients app, installing it using pods thats when I faced those issues (both with swift4 and now with swift4.2) I saw an open issue on this repo as well, I think that this will resolve that problem. I'm happy to contribute.

raulriera commented:
What version of iOS are you targeting? As you can see from the sample project, it compiles fine. Can you post the code and not a screenshot of the error? An actual project I can look at

raulriera commented:
It seems like the early migrators screwed something up, but its not a compile error I can easily just change it to this 
https://github.com/raulriera/TextFieldEffects/commit/6eb62bc74e1d0f7abb7e94f86a3d515bfda31d2c

besarism commented:
Ok so exactly, beside (borderLayer.lineCap) and methods there is no other error, other changes for me are just swift style, so I changed the pull request title to  "Swift short code style" for some cases, for example on "UIView.AnimationOptions.beginFromCurrentState" I have changed to ".beginFromCurrentState" or other example to  ".UITextFieldTextDidEndEditing". Thanks.
